src/Storages/StorageS3Settings.cpp

```diff
@@ -137,6 +145,13 @@ void S3Settings::RequestSettings::PartUploadSettings::validate()
             "Setting upload_part_size_multiply_factor is too big ({}). Multiplication to max_upload_part_size ({}) will cause integer overflow",
             ReadableSize(max_part_number), ReadableSize(max_part_number_limit));
 
+    std::unordered_set<String> storage_class_names {"NOT_SET", "STANDARD", "INTELLIGENT_TIERING"};
```

Comment:
May be support all of them as it is not difficult?
Our was library allows to set these: "STANDARD", "REDUCED_REDUNDANCY", "STANDARD_IA", "INTELLIGENT_TIERING", "DEEP_ARCHIVE", "ONEZONE_IA", "GLACIER"